consensus/poet/core/sawtooth_poet/poet_consensus/poet_block_publisher.py

```diff
@@ -233,7 +233,10 @@ def _handle_registration_timeout(self, block_header, poet_enclave_module,
                          'committed by block %s. Create new registration',
                          poet_public_key,
                          block_header.previous_block_id)
-            del self._poet_key_state_store[poet_public_key]
+            try:
+                del self._poet_key_state_store[poet_public_key]
+            except IndexError:
```

Comment:
Doesn't this throw a `KeyError`?

Reply:
good point. the delete method actually eats that exception anyway. I'll pull it out.